lib/src/tasks/test/api.dart

```diff
@@ -18,17 +18,30 @@ import 'dart:async';
 
 import 'package:dart_dev/util.dart' show TaskProcess;
 
+import 'package:dart_dev/src/tasks/serve/api.dart' as serve;
 import 'package:dart_dev/src/tasks/task.dart';
 
-TestTask test(
-    {int concurrency,
+Future<TestTask> test({int concurrency,
+    bool pubServe,
     List<String> platforms: const [],
-    List<String> tests: const []}) {
+    List<String> tests: const []}) async {
+  serve.PubServeTask pubServeTask;
+  int pubPort = null;
+  if (pubServe) {
+    // Start `pub serve` on the `test` directory
+    pubServeTask = serve.pubServe(additionalArgs: ['test']);
+    var serveInfo = await pubServeTask.serveInfos.first;
```

Comment:
Yeah at this point, that's probably what I would do. You could break it up into two different parts - the logs up until the server is ready, and the remainder. That way you could print the first part immediately and the rest of it when the server is stopped.